pyrokinetics/gk_code/GKOutputReaderTGLF.py

```diff
@@ -0,0 +1,383 @@
+import numpy as np
+import xarray as xr
+from typing import Tuple, Dict, Any, Optional
+from pathlib import Path
+
+from .GKOutputReader import GKOutputReader
+from .GKInputTGLF import GKInputTGLF
+from ..typing import PathLike
+
+
+class TGLFFile:
+    def __init__(self, path: PathLike, required: bool):
+        self.path = Path(path)
+        self.required = required
+        self.fmt = self.path.name.split(".")[0]
+
+
+class GKOutputReaderTGLF(GKOutputReader):
+
+    fields = ["phi", "apar", "bpar"]
+
+    @staticmethod
+    def _required_files(dirname: PathLike):
+        dirname = Path(dirname)
+        return {
+            "input": TGLFFile(dirname / "input.tglf", required=True),
+            "run": TGLFFile(dirname / "out.tglf.run", required=True),
+        }
+
+    def verify(self, dirname: PathLike):
+        dirname = Path(dirname)
+        for f in self._required_files(dirname).values():
+            if not f.path.exists():
+                raise RuntimeError
```

Comment:
If I'm honest I just copied this over from `GKOutputReaderCGYRO`. @LiamPattinson  thoughts?

Reply:
The `verify` function is mostly used for file-type inference, so the raised exception is just there to tell the output reader factory, "This path doesn't look like a directory containing TGLF files, try a different output reader". It might be worth adding more precise error messages to the verify functions throughout Pyrokinetics in case users find another use for them, but I don't think it's a high priority.

Ideally, we could have something like the following for each class with a verify function:
```python
# Maybe @staticmethod?
def verify(self, path, throw=True):
    try:
        # check that the file is valid, raise exception otherwise
        return True
    except Exception as exc:
        if throw:
            raise VerifyPathException(
                f"The path {path} is not a ... file"
            ) from exc
        return False
```